src/known_key.rs

```diff
@@ -0,0 +1,245 @@
+            .and_then(|m| m.raw_entry().from_key_hashed_nocheck(self.hash, &self.key))
+            .map(|kv| kv.1)
+    }
+
+    /// Looks up this key in a `BorrowedValue`, returns None if the
+    /// key wasn't present or `target` isn't an object
+    #[inline]
+    pub fn lookup_mut<'borrow, 'value>(
+        &self,
+        target: &'borrow mut BorrowedValue<'value>,
+    ) -> Option<&'borrow mut BorrowedValue<'value>>
+    where
+        'key: 'value,
+        'value: 'borrow,
+    {
+        target.as_object_mut().and_then(|m| {
+            match m
+                .raw_entry_mut()
+                .from_key_hashed_nocheck(self.hash, &self.key)
+            {
+                RawEntryMut::Occupied(e) => Some(e.into_mut()),
+                RawEntryMut::Vacant(_e) => None,
+            }
+        })
+    }
+
+    /// Looks up this key in a `BorrowedValue`, inserts `with` when the key
+    ///  when wasn't present returns None if the `target` isn't an object
+    #[inline]
+    pub fn lookup_or_insert_mut<'borrow, 'value, F>(
+        &self,
+        target: &'borrow mut BorrowedValue<'value>,
+        with: F,
+    ) -> Result<&'borrow mut BorrowedValue<'value>, Error>
+    where
+        'key: 'value,
+        'value: 'borrow,
+        F: FnOnce() -> BorrowedValue<'value>,
+    {
+        if !target.is_object() {
+            return Err(Error::NotAnObject(target.value_type()));
+        }
+        target
+            .as_object_mut()
+            .map(|m| {
+                m.raw_entry_mut()
+                    .from_key_hashed_nocheck(self.hash, &self.key)
+                    .or_insert_with(|| (self.key.clone(), with()))
+                    .1
+            })
+            .ok_or(Error::NotAnObject(ValueType::Null))
+    }
+
+    /// Inserts a value key into  `BorrowedValue`, returns None if the
+    /// key wasn't present otherwise Some(`old value`).
+    /// Errors if `target` isn't an object
+    #[inline]
+    pub fn insert<'borrow, 'value>(
+        &self,
+        target: &'borrow mut BorrowedValue<'value>,
+        value: BorrowedValue<'value>,
+    ) -> Result<Option<BorrowedValue<'value>>, Error>
+    where
+        'key: 'value,
+        'value: 'borrow,
+    {
+        if !target.is_object() {
+            return Err(Error::NotAnObject(target.value_type()));
+        }
+
+        Ok(target.as_object_mut().and_then(|m| {
+            match m
+                .raw_entry_mut()
+                .from_key_hashed_nocheck(self.hash, &self.key)
+            {
+                RawEntryMut::Occupied(mut e) => Some(e.insert(value)),
+                RawEntryMut::Vacant(e) => {
+                    e.insert_hashed_nocheck(self.hash, self.key.clone(), value);
+                    None
+                }
+            }
+        }))
+    }
+}
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::unnecessary_operation, clippy::non_ascii_literal)]
+    use super::*;
+    use crate::borrowed::*;
+    use crate::ValueTrait;
+
```

Comment:
Maybe an end-to-end test somewhere (perhaps with a doc comment with usage somewhere in the API?) would make things a bit clearer? These tests avoid the JSON processing APIs (targeted & good for a unit test, my brain was looking for an illustrative integration test(s)? I dunno...)

Reply:
added!